mobile-widgets/qml/CloudCredentials.qml

```diff
@@ -97,6 +97,27 @@ Item {
 				onCheckedChanged: {
 					password.echoMode = checked ? TextInput.Normal : TextInput.Password
 				}
+				indicator: Rectangle {
+					implicitWidth: Kirigami.Units.largeSpacing * 3
+					implicitHeight: Kirigami.Units.largeSpacing
+					x: showPassword.leftPadding
+					y: parent.height / 2 - height / 2
+					radius: Kirigami.Units.largeSpacing * 0.5
+					color: showPassword.checked ?
+						subsurfaceTheme.lightPrimaryColor : subsurfaceTheme.backgroundColor
+					border.color: subsurfaceTheme.darkerPrimaryColor
+
+					Rectangle {
+						x: showPassword.checked ? parent.width - width : 0
+						y: parent.height / 2 - height / 2
+						width: Kirigami.Units.largeSpacing * 1.5
+						height: Kirigami.Units.largeSpacing * 1.5
+						radius: height / 2
+						color: showPassword.down || showPassword.checked ?
+							subsurfaceTheme.primaryColor : subsurfaceTheme.lightPrimaryColor
+						border.color: subsurfaceTheme.darkerPrimaryColor
+					}
+				}
```

Comment:
I said this earlier already... we really should just have a StyledCheckbox.qml or something and use that. It would make the code so much simpler and easier to read.